From: Laurent Bigonville <bigon@bigon.be>
Date: Mon, 22 Nov 2021 13:03:48 +0100
Subject: Revert "gedit-dirs: remove get_user_cache_dir() (no longer used)"

This also revert the move of the metadata file ("gedit-metadata.xml")
from user_data_dir to user_cache_dir when not using GVFS to store the
metadata

This reverts commit f641a246fbaff482e216197d21e2e51397b1cb26.

Fixes: https://gitlab.gnome.org/GNOME/gedit/-/issues/467
(cherry picked from commit ba2662ff9ee99ef0d85023b06053792fd3eb9272)
(cherry picked from commit 741be1b11b977abd529aa2f633e50c2e80864afc)
---
 gedit/gedit-dirs.c | 11 +++++++++++
 gedit/gedit-dirs.h |  2 ++
 2 files changed, 13 insertions(+)

diff --git a/gedit/gedit-dirs.c b/gedit/gedit-dirs.c
index 964e0a8..91a4ce4 100644
--- a/gedit/gedit-dirs.c
+++ b/gedit/gedit-dirs.c
@@ -26,6 +26,7 @@
 #include <gtkosxapplication.h>
 #endif
 
+static gchar *user_cache_dir         = NULL;
 static gchar *user_config_dir        = NULL;
 static gchar *user_data_dir          = NULL;
 static gchar *user_styles_dir        = NULL;
@@ -96,6 +97,9 @@ gedit_dirs_init ()
 							   NULL);
 	}
 
+	user_cache_dir = g_build_filename (g_get_user_cache_dir (),
+					   "gedit",
+					   NULL);
 	user_config_dir = g_build_filename (g_get_user_config_dir (),
 					    "gedit",
 					    NULL);
@@ -119,6 +123,7 @@ gedit_dirs_init ()
 void
 gedit_dirs_shutdown ()
 {
+	g_clear_pointer (&user_cache_dir, g_free);
 	g_clear_pointer (&user_config_dir, g_free);
 	g_clear_pointer (&user_data_dir, g_free);
 	g_clear_pointer (&user_styles_dir, g_free);
@@ -130,6 +135,12 @@ gedit_dirs_shutdown ()
 	g_clear_pointer (&gedit_plugins_data_dir, g_free);
 }
 
+const gchar *
+gedit_dirs_get_user_cache_dir (void)
+{
+	return user_cache_dir;
+}
+
 const gchar *
 gedit_dirs_get_user_config_dir (void)
 {
diff --git a/gedit/gedit-dirs.h b/gedit/gedit-dirs.h
index 7f0126c..acd2424 100644
--- a/gedit/gedit-dirs.h
+++ b/gedit/gedit-dirs.h
@@ -32,6 +32,8 @@ void		 gedit_dirs_init			(void);
 void		 gedit_dirs_shutdown 			(void);
 
 
+const gchar	*gedit_dirs_get_user_cache_dir		(void);
+
 const gchar	*gedit_dirs_get_user_config_dir		(void);
 
 const gchar	*gedit_dirs_get_user_data_dir		(void);
